packages/core/test/test-util/test-services/openapi/test-service/model/complex-test-entity.ts

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright (c) 2021 SAP SE or an SAP affiliate company. All rights reserved.
+ *
+ * This is a generated file powered by the SAP Cloud SDK for JavaScript.
+ */
+      import { SimpleTestEntity } from './simple-test-entity';
+      import { TestEntity } from './test-entity';
+      import { Except } from '@sap-cloud-sdk/core';
+      export type ComplexTestEntity = {
+      'referenceProperty'?: SimpleTestEntity;
+      'arrayProperty'?: {
+            'item'?: string;
+          } | Record<string, any>[];
+      'uniqueItemsProperty'?: Set<string>;
+      'requiredPropertiesProperty'?: {
+            'optionalProperty'?: string;
+            'requiredProperty': string;
+          } | Record<string, any>;
+      'enumProperty'?: 'one' | 'two';
+      'oneOfProperty'?: SimpleTestEntity | TestEntity;
+      'allOfProperty'?: SimpleTestEntity & {
```

Comment:
this type might be too tolerant? Because it accepts {abc: ''}. I guess you have special reasons for `| Record<string, any>`?

Reply:
Additional properties are only used in objects, that is correct. In this case, the type is correct, because one of the types given is an object.